sstables/storage.cc

```diff
@@ -461,21 +461,20 @@ class s3_storage : public sstables::storage {
 };
 
 sstring s3_storage::make_s3_object_name(const sstable& sst, component_type type) {
-    return format("/{}/{}/{}", _bucket, *_remote_prefix, sstable_version_constants::get_component_map(sst.get_version()).at(type));
+    return format("/{}/{}/{}", _bucket, sst.generation(), sstable_version_constants::get_component_map(sst.get_version()).at(type));
```

Comment:
If you use the `sst.generation()` here explicitly, there's no need in keeping the `_remote_prefix` thing on board

Reply:
i am confused.  as the title puts, the whole point of this commit is to remove `_remote_prefix`.

Reply:
Yes, that's my point -- the `_remote_prefix` is still here in this patch, but I then found that you remove it in the _second_ one :) So this concern is invalid